src/diagnostics/radiation_diagnostics.jl

```diff
@@ -39,6 +39,93 @@ add_diagnostic_variable!(
     compute! = compute_rsd!,
 )
 
+###
+# TOA downwelling shortwave radiation (3d)
+###
+compute_rsdt!(out, state, cache, time) =
+    compute_rsdt!(out, state, cache, time, cache.atmos.radiation_mode)
+compute_rsdt!(_, _, _, _, radiation_mode::T) where {T} =
+    error_diagnostic_variable("rsdt", radiation_mode)
+
+function compute_rsdt!(
+    out,
+    state,
+    cache,
+    time,
+    radiation_mode::T,
+) where {T <: RRTMGPI.AbstractRRTMGPMode}
+    nlevels = Spaces.nlevels(axes(state.c))
+    if isnothing(out)
+        return Fields.level(
+            RRTMGPI.array2field(
+                cache.radiation.radiation_model.face_sw_flux_dn,
+                axes(state.f),
+            ),
+            nlevels + half,
+        )
+    else
+        out .= Fields.level(
+            RRTMGPI.array2field(
+                cache.radiation.radiation_model.face_sw_flux_dn,
+                axes(state.f),
+            ),
+            nlevels + half,
+        )
+    end
```

Comment:
```suggestion
    face_sw_flux_dn_field = RRTMGPI.array2field(cache.radiation.radiation_model.face_sw_flux_dn, axes(state.f))
    lev = Fields.level(face_sw_flux_dn_field, nlevels + half)
    if isnothing(out)
        return lev
    else
        out .= lev
    end
```
Can we break down the expression like this?

Reply:
Yes. Why would this work though?

Reply:
Ah, I don't think it works: https://buildkite.com/clima/climaatmos-ci/builds/16245#018d537a-779d-4403-ac51-0bfea09917d1/163-260

Reply:
It might not, but it could help narrow down the issue.

Reply:
The problem is described below. It is the `copy` function with called with a particular combination of `view/reshape/view`. This only occurs in this case because of the different DataLayout.